presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -408,7 +408,9 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
 
     private Predicate<NullableValue> convertToPredicate(TupleDomain<ColumnHandle> predicate, TpchColumn column)
     {
-        return nullableValue -> predicate.contains(TupleDomain.fromFixedValues(ImmutableMap.of(toColumnHandle(column), nullableValue)));
+        TpchColumnHandle columnHandle = toColumnHandle(column);
+        TupleDomain<ColumnHandle> columnPredicate = filterColumns(predicate, columnHandle::equals);
+        return nullableValue -> columnPredicate.contains(TupleDomain.fromFixedValues(ImmutableMap.of(columnHandle, nullableValue)));
```

Comment:
it took me a while to understand this issue. Maybe we could provide tests for that. Or we could leave https://github.com/prestodb/presto/pull/9823 to cover this with ATQ

Reply:
i removed the commit from this PR.

Reply:
oh, i didn't, it was a required commit